### PR TITLE
replace current tiflash with tiflash cr

### PR DESCRIPTION
**Pull request by yeya24**

### CODE REVIEW

File: pkg/test-infra/tidb/operation.go [hunk 1]
```diff
@@ -63,8 +63,8 @@ type Ops struct {
 }
 
 // New ...
-func New(namespace, name string, config fixture.TiDBClusterConfig) *Ops {
-	return &Ops{cli: tests.TestClient.Cli, tc: RecommendedTiDBCluster(namespace, name, config),
+func New(namespace, name string, config fixture.TiDBClusterConfig, enableTiFlash bool) *Ops {
```
Comment (mahjonp) [inline pkg/test-infra/tidb/operation.go:66]: How about putting enableTiFlash into fixture.TiDBClusterConfig and using TiFlashReplicas to control whether or not require tiflash?

Comment (yeya24) [reply pkg/test-infra/tidb/operation.go:66]: Yes, that is a good idea

**Comment by yeya24** [comment]
Updated

File: pkg/test-infra/tidb/recommend.go [hunk 2]
```diff
@@ -220,11 +249,23 @@ func RecommendedTiDBCluster(ns, name string, clusterConfig fixture.TiDBClusterCo
...
 			},
 		},
 	}
 
+	if clusterConfig.TiFlashReplicas > 0 {
+		r.EnableTiFlash(clusterConfig.TiFlashImageVersion, clusterConfig.TiFlashReplicas)
+
+		// TODO(yeya24): There is a bug in the operator side so we can only set it
+		// manually https://github.com/pingcap/tidb-operator/issues/2219.
+		// Remove this after it fixes the bug.
```
Comment (mahjonp) [inline pkg/test-infra/tidb/recommend.go:261]: So it's ok to use tidb-operator with TiFlash now? I see DanielZhangQD said that it's still in progress?

Comment (yeya24) [reply pkg/test-infra/tidb/recommend.go:261]: For deploying tiflash it is okay now. WIP is meant for scale in and scale out, which is not necessary in tipocket

**Pull request merged**